src/test-run/index.js

```diff
@@ -270,13 +272,43 @@ export default class TestRun extends Session {
         return this.currentDriverTask ? this.currentDriverTask.command : null;
     }
 
+    static _transformResizeWindowToFitDeviceCommand (command) {
+        var { landscapeWidth, portraitWidth } = getViewportSize(command.device);
+        var portrait = command.options.portraitOrientation;
+        var width    = portrait ? portraitWidth : landscapeWidth;
+        var height   = portrait ? landscapeWidth : portraitWidth;
+
+        return new ResizeWindowCommand({ width, height });
+    }
+
+    _isValidResize (command) {
+        if (!OS.mac)
+            return true;
+
+        var providerName  = this.browserConnection.browserInfo.providerName;
+        var browserId     = this.browserConnection.id;
+        var provider      = this.browserConnection.provider;
+        var maxScreenSize = provider.maxScreenSizes[browserId];
+
+        if (providerName !== 'remote' && providerName !== 'path' && providerName !== 'locally-installed')
```

Comment:
Why? If I create provider for redistributable chrome that runs on macOs this will not work?